riscv-page-tables/src/page_table.rs

```diff
@@ -633,34 +633,44 @@ impl<T: PagingMode> PageTableInner<T> {
         }
     }
 
-    /// Locks the invalid leaf PTE mapping `vaddr`, filling in any missing intermediate page tables
-    /// using `get_pte_page`.
-    fn lock_leaf_for_mapping(
+    /// Locks the invalid leaf PTEs mappings starting from `vaddr` for `num_pages` pages of size
+    /// `page_size`, filling in any missing intermediate page tables using `get_pte_page`.
+    fn lock_leaves_for_mapping(
         &mut self,
-        vaddr: PageAddr<T::MappedAddressSpace>,
+        start_addr: PageAddr<T::MappedAddressSpace>,
+        num_pages: u64,
         page_size: PageSize,
         get_pte_page: &mut dyn FnMut() -> Option<Page<InternalClean>>,
     ) -> Result<()> {
-        let mut table = PageTable::from_root(self);
-        // match the level to the page size
-        while table.level.leaf_page_size() != page_size {
-            table = table.next_level_or_fill_fn(RawAddr::from(vaddr), get_pte_page)?;
-        }
-        let entry = table.entry_for_addr_mut(RawAddr::from(vaddr));
-        use TableEntryType::*;
-        match entry {
-            Invalidated(i) => {
-                i.lock();
-                Ok(())
+        start_addr
+            .checked_add_pages_with_size(num_pages, page_size)
+            .ok_or(Error::OutOfMapRange)?;
+
+        let iterator = start_addr
+            .iter_from_with_size(page_size)
+            .ok_or_else(|| Error::AddressMisaligned(start_addr.bits()))?;
+
+        for vaddr in iterator.take(num_pages as usize) {
```

Comment:
nit: don't need to create an `iterator` variable:
```rust
for vaddr in start_addr.iter_from_with_size(page_size).ok_or_else(|| Error::AddressMisaligned(start_addr.bits()))?.take(num_pages as usize) {
```

Reply:
Of course. This is temporary and I am just trying to make the iteration explicit.